src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/DeleteBookingTest.kt

```diff
@@ -0,0 +1,151 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import com.ninjasquad.springmockk.SpykBean
+import io.mockk.every
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.springframework.data.repository.findByIdOrNull
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ExtensionRepository
+
+class DeleteBookingTest : IntegrationTestBase() {
+  @SpykBean
+  lateinit var realExtensionRepository: ExtensionRepository
+
+  @Test
+  fun `Deleting a Booking successfully deletes all related entities and deletes the Booking itself`() {
+    val premises = approvedPremisesEntityFactory.produceAndPersist {
```

Comment:
I think we just need premises to be set up for us to enable the rest of the bookings to be set up? If so, I'd probably reach to putting this into a reusable helper. It could give others the steer and encourage a single definition and help the reader skip to the essential bits. Just an idea, no action needed from me 👍